azkaban-common/src/main/java/azkaban/project/JdbcProjectImpl.java

```diff
@@ -1015,31 +1017,37 @@ public void uploadFlowFile(final int projectId, final int projectVersion, final
 
   @Override
   public File getUploadedFlowFile(final int projectId, final int projectVersion,
-      final int flowVersion, final String flowName) throws ProjectManagerException {
+      final String flowFileName,
+      final int flowVersion) throws ProjectManagerException, IOException {
     final FlowFileResultHandler handler = new FlowFileResultHandler();
 
     final List<byte[]> data;
-    // Todo jamiesjc: delete the flow file after used.
-    final File file = new File(this.tempDir, flowName);
+    final File tempDir = Utils.createTempDir(this.tempDir);
+    final File file = new File(tempDir, flowFileName);
```

Comment:
would be good to add some comments explaining the motivation of creating a temp dir.

Reply:
when will the temp dir get deleted? It's not a good idea to just delete on exit since linux has hard limit on the number of dir/file contained in a dir.

Reply:
Refactored some of the code and handled the deletion of temp dir.